src/frontend/src/eth/components/transactions/Transaction.svelte

```diff
@@ -2,43 +2,84 @@
 	import { nonNullish } from '@dfinity/utils';
 	import { BigNumber } from '@ethersproject/bignumber';
 	import type { ComponentType } from 'svelte';
-	import type { EthTransactionType } from '$eth/types/eth-transaction';
+	import type { Erc20Token } from '$eth/types/erc20';
+	import type { EthTransactionType, EthTransactionUi } from '$eth/types/eth-transaction';
+	import { isSupportedEthTokenId } from '$eth/utils/eth.utils';
 	import { isTransactionPending } from '$eth/utils/transactions.utils';
+	import IconBurn from '$lib/components/icons/IconBurn.svelte';
+	import IconConvert from '$lib/components/icons/IconConvert.svelte';
+	import IconMint from '$lib/components/icons/IconMint.svelte';
 	import IconReceive from '$lib/components/icons/IconReceive.svelte';
 	import IconSend from '$lib/components/icons/IconSend.svelte';
 	import TransactionPending from '$lib/components/transactions/TransactionPending.svelte';
 	import Amount from '$lib/components/ui/Amount.svelte';
 	import Card from '$lib/components/ui/Card.svelte';
 	import RoundedIcon from '$lib/components/ui/RoundedIcon.svelte';
-	import { ethAddress } from '$lib/derived/address.derived';
+	import { tokenId, tokenSymbol } from '$lib/derived/token.derived';
 	import { i18n } from '$lib/stores/i18n.store';
 	import { modalStore } from '$lib/stores/modal.store';
-	import type { Transaction } from '$lib/types/transaction';
+	import { token } from '$lib/stores/token.store';
 	import { formatSecondsToDate } from '$lib/utils/format.utils';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
 
-	export let transaction: Transaction;
+	export let transaction: EthTransactionUi;
 
-	let from: string;
 	let value: BigNumber;
 	let timestamp: number | undefined;
 	let displayTimestamp: number | undefined;
+	let type: EthTransactionType;
 
 	let pending: boolean;
 	$: pending = isTransactionPending(transaction);
 
-	$: ({ from, value, timestamp, displayTimestamp } = transaction);
+	$: ({ value, timestamp, displayTimestamp, uiType: type } = transaction);
 
-	let type: EthTransactionType;
-	$: type = from?.toLowerCase() === $ethAddress?.toLowerCase() ? 'send' : 'receive';
+	let ckTokenSymbol: string;
+	$: ckTokenSymbol =
+		nonNullish($tokenId) && isSupportedEthTokenId($tokenId)
+			? `ck${$tokenSymbol}`
```

Comment:
Not sure why this `ck${$tokenSymbol}` mapping, can you briefly explain what information we are missing?

Reply:
I think there is something, like not a link but there is already a piece of code somewhere that does something similar....

Reply:
I don't remember where so not a blocker but, ....

Reply:
I searched and I cannot find anything, do you have some clue on what to search? Or maybe about which flow?

P.S. the alternative would be to make an array of ckETH data (prod, staging and local) and use `.find` to compare against their `twinToken` property. That would be the completly correct way

Reply:
```typescript
// PSEUDO CODE

export const CKETH_DATA: IcCkInterface[] = [
	...(nonNullish(CKETH_LOCAL_DATA) ? [CKETH_LOCAL_DATA] : []),
	...(nonNullish(CKETH_STAGING_DATA) ? [CKETH_STAGING_DATA] : []),
	...(nonNullish(CKETH_IC_DATA) ? [CKETH_IC_DATA] : [])
];

CKETH_DATA.find(({ twinToken: { id } }) => id === $tokenId)?.symbol;
```

Reply:
Ah, couldn't we use `twinTokenSymbol` or `Erc20Contract`?

Reply:
for ETH and SepoliaETH we don't have the property `twinTokenSymbol`, we never assign it. Shall I add it (in another PR)?

Reply:
Doing with PRs:

- https://github.com/dfinity/oisy-wallet/pull/2348
- https://github.com/dfinity/oisy-wallet/pull/2350
-